src/main/java/com/synopsys/integration/blackduck/codelocation/signaturescanner/command/ScanCommandCallable.java

```diff
@@ -58,6 +60,13 @@ public ScanCommandOutput call() {
             cmd.add(scanCommand.getTargetPath());
 
             commandToExecute = createPrintableCommand(cmd);
+            
+            if (OperatingSystemType.determineFromSystem().equals(OperatingSystemType.WINDOWS) 
+            		&& commandToExecute.length() > WINDOWS_CHARACTER_LIMIT) {
+        	    String errorMessage = "Unable to invoke the scan CLI as the length of the command would exceed the operating system limit.";
+        	    return ScanCommandOutput.FAILURE(projectAndVersion, codeLocationName, logger, scanCommand, commandToExecute, errorMessage, null);              
+            }
+           
```

Comment:
Do we want to guard against other operating systems or is this only happening with Windows?

Reply:
Only Windows.

Reply:
It's a good question. Windows runs into trouble around 32k. I believe mac and linux run into trouble at around 200k. Given that's pretty huge I didn't want to restrict things there and potentially run into trouble but I could certainly add a check if we think it worthwhile.

Reply:
I'm fine with merging these changes and addressing issues with linux/Mac if and when that is reported